engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java

```diff
@@ -188,10 +243,115 @@ private void recursivelyAddModuleDependencies(Set<Name> modules, Name moduleName
+        context.put(UniverseSetupScreen.class, this);
+        assetTypeManager = new ModuleAwareAssetTypeManager();
+        context.put(AssetManager.class, assetTypeManager.getAssetManager());
+        context.put(ModuleAwareAssetTypeManager.class, assetTypeManager);
+        context.put(ModuleManager.class, moduleManager);
+        DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+        context.put(UniverseWrapper.class, wrapper);
+        ResolutionResult result = resolver.resolve(config.getDefaultModSelection().listModules());
+
+        if (result.isSuccess()) {
+            environment = moduleManager.loadEnvironment(result.getModules(), false);
+            context.put(ModuleEnvironment.class, environment);
+            context.put(WorldGeneratorPluginLibrary.class, new TempWorldGeneratorPluginLibrary(environment, context));
+            initAssets();
+
+            EnvironmentSwitchHandler environmentSwitcher = new EnvironmentSwitchHandler();
+            context.put(EnvironmentSwitchHandler.class, environmentSwitcher);
+
+            environmentSwitcher.handleSwitchToPreviewEnvironment(context, environment);
+        }
+    }
+
+    private void initAssets() {
+        DefaultBlockFamilyFactoryRegistry familyFactoryRegistry = new DefaultBlockFamilyFactoryRegistry();
+        context.put(BlockFamilyFactoryRegistry.class, familyFactoryRegistry);
+
+        // cast lambdas explicitly to avoid inconsistent compiler behavior wrt. type inference
+        assetTypeManager.registerCoreAssetType(Prefab.class,
+                (AssetFactory<Prefab, PrefabData>) PojoPrefab::new, false, "prefabs");
+        assetTypeManager.registerCoreAssetType(BlockShape.class,
+                (AssetFactory<BlockShape, BlockShapeData>) BlockShapeImpl::new, "shapes");
+        assetTypeManager.registerCoreAssetType(BlockSounds.class,
+                (AssetFactory<BlockSounds, BlockSoundsData>) BlockSounds::new, "blockSounds");
+        assetTypeManager.registerCoreAssetType(BlockTile.class,
+                (AssetFactory<BlockTile, TileData>) BlockTile::new, "blockTiles");
+        assetTypeManager.registerCoreAssetType(BlockFamilyDefinition.class,
+                (AssetFactory<BlockFamilyDefinition, BlockFamilyDefinitionData>) BlockFamilyDefinition::new, "blocks");
+        assetTypeManager.registerCoreFormat(BlockFamilyDefinition.class,
+                new BlockFamilyDefinitionFormat(assetTypeManager.getAssetManager(), familyFactoryRegistry));
+        assetTypeManager.registerCoreAssetType(UISkin.class,
+                (AssetFactory<UISkin, UISkinData>) UISkin::new, "skins");
+        assetTypeManager.registerCoreAssetType(BehaviorTree.class,
+                (AssetFactory<BehaviorTree, BehaviorTreeData>) BehaviorTree::new, false, "behaviors");
+        assetTypeManager.registerCoreAssetType(UIElement.class,
+                (AssetFactory<UIElement, UIData>) UIElement::new, "ui");
+
```

Comment:
Extra newline. You don't usually leave a newline at the end of methods, so don't do it here either.